src/code/z_actor.c

```diff
@@ -3646,187 +3646,287 @@ Hilite* func_8003435C(Vec3f* object, PlayState* play) {
-    sp44 = Math_Vec3f_Yaw(&sp30, &arg1->unk_18);
-    sp40 = Math_Vec3f_Yaw(&actor->world.pos, &arg1->unk_18) - actor->shape.rot.y;
+    pitchTowardsPlayer = Math_Vec3f_Pitch(&offsetActorPos, &interactInfo->playerPosition);
+    yawTowardsPlayer = Math_Vec3f_Yaw(&offsetActorPos, &interactInfo->playerPosition);
+    bodyYawDiff = Math_Vec3f_Yaw(&actor->world.pos, &interactInfo->playerPosition) - actor->shape.rot.y;
 
-    temp1 = CLAMP(sp40, -arg2, arg2);
-    Math_SmoothStepToS(&arg1->unk_08.y, temp1, 6, 2000, 1);
+    temp = CLAMP(bodyYawDiff, -maxHeadYaw, maxHeadYaw);
+    Math_SmoothStepToS(&interactInfo->rotHead.y, temp, 6, 2000, 1);
 
-    temp1 = (ABS(sp40) >= 0x8000) ? 0 : ABS(sp40);
-    arg1->unk_08.y = CLAMP(arg1->unk_08.y, -temp1, temp1);
+    temp = (ABS(bodyYawDiff) >= 0x8000) ? 0 : ABS(bodyYawDiff);
+    interactInfo->rotHead.y = CLAMP(interactInfo->rotHead.y, -temp, temp);
 
-    sp40 -= arg1->unk_08.y;
+    bodyYawDiff -= interactInfo->rotHead.y;
 
-    temp1 = CLAMP(sp40, -arg5, arg5);
-    Math_SmoothStepToS(&arg1->unk_0E.y, temp1, 6, 2000, 1);
+    temp = CLAMP(bodyYawDiff, -maxTorsoYaw, maxTorsoYaw);
+    Math_SmoothStepToS(&interactInfo->rotTorso.y, temp, 6, 2000, 1);
 
-    temp1 = (ABS(sp40) >= 0x8000) ? 0 : ABS(sp40);
-    arg1->unk_0E.y = CLAMP(arg1->unk_0E.y, -temp1, temp1);
+    temp = (ABS(bodyYawDiff) >= 0x8000) ? 0 : ABS(bodyYawDiff);
+    interactInfo->rotTorso.y = CLAMP(interactInfo->rotTorso.y, -temp, temp);
 
-    if (arg8) {
-        Math_SmoothStepToS(&actor->shape.rot.y, sp44, 6, 2000, 1);
+    if (rotateActorShape) {
+        Math_SmoothStepToS(&actor->shape.rot.y, yawTowardsPlayer, 6, 2000, 1);
     }
 
-    temp1 = CLAMP(sp46, arg4, (s16)(u16)arg3);
-    Math_SmoothStepToS(&arg1->unk_08.x, temp1, 6, 2000, 1);
+    temp = CLAMP(pitchTowardsPlayer, minHeadPitch, (s16)(u16)maxHeadPitch);
+    Math_SmoothStepToS(&interactInfo->rotHead.x, temp, 6, 2000, 1);
 
-    temp2 = sp46 - arg1->unk_08.x;
+    torsoPitch = pitchTowardsPlayer - interactInfo->rotHead.x;
 
-    temp1 = CLAMP(temp2, arg7, arg6);
-    Math_SmoothStepToS(&arg1->unk_0E.x, temp1, 6, 2000, 1);
+    temp = CLAMP(torsoPitch, minTorsoPitch, maxTorsoPitch);
+    Math_SmoothStepToS(&interactInfo->rotTorso.x, temp, 6, 2000, 1);
 }
 
-s16 func_800347E8(s16 arg0) {
-    return D_80116130[arg0].unk_14;
+s16 Npc_GetPlayerTrackingPresetMaxYaw(s16 presetIndex) {
+    return sPlayerTrackingPresets[presetIndex].maxYawForPlayerTracking;
 }
 
-s16 func_80034810(Actor* actor, struct_80034A14_arg1* arg1, f32 arg2, s16 arg3, s16 arg4) {
+/**
+ * Handles player tracking options and auto-turning towards the player when
+ * NPC_PLAYER_TRACKING_AUTO_TURN tracking option is used.
+ *
+ * Returns a tracking option that will determine which actor limbs
+ * will be rotated towards the player.
+ *
+ * When the player is behind the actor (yaw difference from actor's shape rotation
+ * to the yaw towards player is larger than maxYawForPlayerTracking), the actor
```

Comment:
this feels a bit verbose/technical, maybe can simplify to "When the player is behind the actor (i.e. not in the yaw range in front of the actor defined by maxYawForPlayerTracking), the actor" ?